rust/util/mod.rs

```diff
@@ -52,3 +52,12 @@ macro_rules! write_joined {
         result
     }};
 }
+
+#[macro_export]
+macro_rules! try_ {
+    {$($stmt:tt)*} => {
+        || -> Result<_, ErrorMessage> {
+            Ok({$($stmt)*})
+        }()
+    };
+}
```

Comment:
The core change: wrap the statements in a lambda that returns Result and immediately call it.